flaml/searcher/blendsearch.py

```diff
@@ -143,20 +144,31 @@ def _init_search(self):
         self._deadline = np.inf
 
     def save(self, checkpoint_path: str):
-        save_object = (self._metric_target, self._search_thread_pool,
-            self._thread_count, self._init_used, self._trial_proposed_by,
-            self._ls_bound_min, self._ls_bound_max, self._result,
-            self._deadline)
+        save_object = self

```

Comment:
Does it mean to save the whole BlendSearch instance?

Reply:
yes, to ensure all the states are saved.